UPGRADE.md

```diff
@@ -36,6 +36,294 @@ starts new workers, which take over from old workers before those old workers
++        set $upstream_upgrade            '';
++        set $upstream_connection         '';
++        set $upstream_scheme             '';
++        set $upstream_uri                '';
++        set $upstream_x_forwarded_for    '';
++        set $upstream_x_forwarded_proto  '';
++        set $upstream_x_forwarded_host   '';
++        set $upstream_x_forwarded_port   '';
++
++        rewrite_by_lua_block {
++            kong.rewrite()
++        }
+ 
+         access_by_lua_block {
+             kong.access()
+         }
+ 
+         proxy_http_version 1.1;
+-        proxy_set_header X-Real-IP $remote_addr;
+-        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+-        proxy_set_header X-Forwarded-Proto $scheme;
+-        proxy_set_header Host $upstream_host;
+-        proxy_set_header Upgrade $upstream_upgrade;
+-        proxy_set_header Connection $upstream_connection;
+-
+-        proxy_pass_header Server;
+-        proxy_pass $upstream_scheme://kong_upstream;
++        proxy_set_header   Host              $upstream_host;
++        proxy_set_header   Upgrade           $upstream_upgrade;
++        proxy_set_header   Connection        $upstream_connection;
++        proxy_set_header   X-Forwarded-For   $upstream_x_forwarded_for;
++        proxy_set_header   X-Forwarded-Proto $upstream_x_forwarded_proto;
++        proxy_set_header   X-Forwarded-Host  $upstream_x_forwarded_host;
++        proxy_set_header   X-Forwarded-Port  $upstream_x_forwarded_port;
++        proxy_set_header   X-Real-IP         $remote_addr;
++        proxy_pass_header  Server;
++        proxy_pass_header  Date;
++        proxy_ssl_name     $upstream_host;
++        proxy_pass         $upstream_scheme://kong_upstream$upstream_uri;
+ 
+         header_filter_by_lua_block {
+             kong.header_filter()
+@@ -136,7 +166,8 @@ server {
+     server_name kong_admin;
+     listen ${{ADMIN_LISTEN}};
+ 
+-    access_log logs/admin_access.log;
++    access_log ${{ADMIN_ACCESS_LOG}};
++    error_log ${{ADMIN_ERROR_LOG}} ${{LOG_LEVEL}};
+ 
+     client_max_body_size 10m;
+     client_body_buffer_size 10m;
+@@ -145,14 +176,19 @@ server {
+     listen ${{ADMIN_LISTEN_SSL}} ssl;
+     ssl_certificate ${{ADMIN_SSL_CERT}};
+     ssl_certificate_key ${{ADMIN_SSL_CERT_KEY}};
+-    ssl_protocols TLSv1 TLSv1.1 TLSv1.2;
++    ssl_protocols TLSv1.1 TLSv1.2;
++
++    ssl_session_cache shared:SSL:10m;
++    ssl_session_timeout 10m;
++    ssl_prefer_server_ciphers on;
++    ssl_ciphers ${{SSL_CIPHERS}};
+ > end
+ 
+     location / {
+         default_type application/json;
+         content_by_lua_block {
+             ngx.header['Access-Control-Allow-Origin'] = '*'
+-            ngx.header['Access-Control-Allow-Credentials'] = 'false'
++
+             if ngx.req.get_method() == 'OPTIONS' then
+                 ngx.header['Access-Control-Allow-Methods'] = 'GET,HEAD,PUT,PATCH,POST,DELETE'
+                 ngx.header['Access-Control-Allow-Headers'] = 'Content-Type'
+
+```
```

Comment:
A link to the GH-hosted version of that diff would be preferable, but GH only allows for diffing 2 branches entirely as far as I could tell. A gist could be an option, but as time passes, the gist could result in an 404 some day...